lib/matplotlib/backends/gtk3_compat.py

```diff
@@ -0,0 +1,34 @@
+import matplotlib
+
+error_msg_gtk3 = "Gtk3 backend requires the installation of pygobject or pgi."
+
+# Import the first library that works from the rcParam list
+# throw ImportError if none works
+for lib in matplotlib.rcParams['backend.gi_preference']:
+    try:
+        gi = __import__(lib, globals(), locals(), [], 0)
```

Comment:
I am not very familiar with the import mechanisms of python. Can you explain why `import lib as gi` wouldn't work here and why not resorting to `importlib` isn't an option? My very little knowledge says that `import` > `importlib` >> `__import__`.
Right now, it looks like we are trying to be too smart about this.